.github/workflows/platform-eol-check.yml

```diff
@@ -0,0 +1,108 @@
+  group: eol-check-${{ github.repository }}
+  cancel-in-progress: true
+jobs:
+  # Prepare the build matrix.
+  # This uses output from .github/scripts/gen-matrix-eol-check.py
+  matrix:
+    name: Prepare Build Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+      - name: Prepare tools
+        id: prepare
+        run: |
+          sudo apt-get update && sudo apt-get install -y python3-ruamel.yaml
+      - name: Read build matrix
+        id: set-matrix
+        run: |
+          matrix="$(.github/scripts/gen-matrix-eol-check.py)"
+          echo "Generated matrix: ${matrix}"
+          echo "matrix=${matrix}" >> "${GITHUB_OUTPUT}"
+
+  eol-check:
+    name: EOL Check
+    runs-on: ubuntu-latest
+    needs:
+      - matrix
+    strategy:
+      matrix: ${{ fromJson(needs.matrix.outputs.matrix) }}
+      fail-fast: false  # We want to check everything, so don’t bail on the first failure.
+      max-parallel: 2  # Cap of two jobs at a time to limit impact on other CI.
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+      # Actually check the EOL date for the platform.
+      - name: Check EOL Date
+        id: check
+        run: |
+          d="$(.github/scripts/platform-impending-eol.py ${{ matrix.distro }} ${{ matrix.release }})"
+          case $? in
+            0) echo "pending=false" >> "${GITHUB_OUTPUT}" ;;
+            2)
+              echo "pending=true" >> "${GITHUB_OUTPUT}"
+              echo "date=${d}" >> "${GITHUB_OUTPUT}"
+              ;;
+            *)
+              echo "::error::Failed to check EOL date for ${{ matrix.distro }} ${{ matrix.release }}"
+              exit 1
+              ;;
+          esac
+      # Figure out the issue title.
+      # This is it’s own step so we only have to set it in one place.
+      - name: Determine Issue Title
+        id: title
+        if: steps.check.outputs.pending == 'true'
+        run: |
+          echo "title=[Platform EOL]: ${{ matrix.full_name }} will be EOL soon." >> "${GITHUB_OUTPUT}"
+      # Check if there is an existing issue in the repo for the platform EOL.
+      # The actual command line to make the check is unfortunately complicated because
+      # GitHub think that it’s sensible to exit with a status of 0 if there no results
+      # for a search.
+      - name: Check for Existing Issue
+        id: existing
+        if: steps.check.outputs.pending == 'true'
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          count=$(gh issue list -R netdata/netdata -s all -S '${{ steps.title.outputs.title }} in:title' --json 'id' -q '. | length')
+          if [ "${count}" -ge 1 ]; then
+            echo 'exists=true' >> "${GITHUB_OUTPUT}"
+          else
+            echo 'exists=false' >> "${GITHUB_OUTPUT}"
+          fi
```

Comment:
Will `exists` be `false` if `gh issue ...` fails (`[ "${count}" -ge 1 ]` fails if count is unset).

Reply:
The job will actually fail in that case due to the shell spitting out an error (it ends up trying to compare a string to a number).

This is actually the behavior we want here, because it makes sure we actually see that something went wrong (and it errs on the side of caution by not creating an issue at all in that case).

Reply:
Are you sure?

```
pve-deb-work ~ $ cat -p qq.sh
#!/bin/bash

if [ "${count}" -ge 1 ]; then
  echo 'exists=true'
else
  echo 'exists=false'
fi
pve-deb-work ~ $ ./qq.sh
./qq.sh: line 3: [: : integer expression expected
exists=false
pve-deb-work ~ $ echo $?
0
```

I did test it in CI and the job doesn't fail.

Reply:
Hmm, I was under the impression that script blocks ran with `set -e` enabled. Apparently not.

Reply:
Adding `-e` doesn't fix the problem (tested in my repo). We could add `[ -z "${count}" ] && exit 1`.

Reply:
Ok, I think I was wrong. The unhandled case: `gh` exit code is 0 but the command output is not a number. I think we can ignore it.